advisor/src/main/resources/META-INF/services/org.ossreviewtoolkit.advisor.AdvisorFactory

```diff
@@ -1 +1,2 @@
 org.ossreviewtoolkit.advisor.advisors.NexusIq$Factory
+org.ossreviewtoolkit.advisor.advisors.VulnerableCode$Factory
```

Comment:
This change should actually be part of the previous commit.

Reply:
Really? For me this is part of the integration of the advisor into the ORT infrastructure, but not part of the implementation itself. The implementation works well (all tests run successful) without this line.

Reply:
The fact that tests work is a red herring, as the test do not use the advisor's ServiceLoader mechanism. With the commits up to (and including) "Add implementation of VulnerableCode advisor", a user would not be able to use the advisor library programmatically with VulnerableCode, as the implementation would not be discoverable. In that sense, yes, adding `org.ossreviewtoolkit.advisor.advisors.VulnerableCode$Factory` to this file for me belongs to the commit that introduces `VulnerableCode.Factory`.

Reply:
The tests are unit tests and therefore test the component in isolation.

Well, we can agree to disagree then. In order to move things forward, I have changed the commits as requested.

Reply:
Thanks. Maybe this argument convinces you more, completely disregarding the tests: Let's assume for a second that it would not have been the `cli` application but a change to the `helper-cli` application that would have used this new feature of the `advisor` library first. Would it then have been the commit in the `helper-cli` application that did this change? In other words, whoever uses that feature from the library first must change the library to actually make it usable? That does not make sense to me, and I believe the library itself needs to be in a state to be usable by anyone; hence that change (which is a change in the library code) belongs to the library, not to the application using the library.